kratos/processes/calculate_embedded_nodal_variable_from_skin_process.h

```diff
@@ -0,0 +1,729 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Ruben Zorrilla
+//
+//
+
+#if !defined(KRATOS_CALCULATE_EMBEDDED_VARIABLE_FROM_SKIN_PROCESS_INCLUDED )
+#define  KRATOS_CALCULATE_EMBEDDED_VARIABLE_FROM_SKIN_PROCESS_INCLUDED
+
+// System includes
+#include <string>
+#include <iostream>
+#include <algorithm>
+
+// External includes
+
+// Project includes
+#include "containers/model.h"
+#include "includes/define.h"
+#include "includes/model_part.h"
+#include "includes/kratos_flags.h"
+#include "elements/embedded_nodal_variable_calculation_element_simplex.h"
+#include "linear_solvers/cg_solver.h"
+#include "processes/process.h"
+#include "processes/find_intersected_geometrical_objects_process.h"
+#include "solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h"
+#include "solving_strategies/schemes/residualbased_incrementalupdate_static_scheme.h"
+#include "solving_strategies/strategies/residualbased_linear_strategy.h"
+#include "utilities/intersection_utilities.h"
+#include "utilities/variable_utils.h"
+
+namespace Kratos
+{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+template< class TVarType >
+class EmbeddedNodalVariableFromSkinTypeHelperClass
+{
+public:
+
+    ///@name Type Definitions
+    ///@{
+
+
+    ///@}
+    ///@name Pointer Definitions
+
+    /// Pointer definition of EmbeddedNodalVariableFromSkinTypeHelperClass
+    KRATOS_CLASS_POINTER_DEFINITION(EmbeddedNodalVariableFromSkinTypeHelperClass);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constructor
+    EmbeddedNodalVariableFromSkinTypeHelperClass() {};
+
+    /// Destructor.
+    ~EmbeddedNodalVariableFromSkinTypeHelperClass() {};
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    static inline const Variable<TVarType> &GetUnknownVariable();
+
+    static inline void AddUnknownVariable(ModelPart &rModelPart);
+
+    static inline void AddUnknownVariableDofs(ModelPart &rModelPart);
+
+    ///@}
+};
+
+template <>
+inline const Variable<double> &EmbeddedNodalVariableFromSkinTypeHelperClass<double>::GetUnknownVariable()
+{
+    return KratosComponents<Variable<double>>::Get("NODAL_MAUX");
+}
+
+template <>
+inline const Variable<array_1d<double,3>> &EmbeddedNodalVariableFromSkinTypeHelperClass<array_1d<double,3>>::GetUnknownVariable()
+{
+    return KratosComponents<Variable<array_1d<double, 3>>>::Get("NODAL_VAUX");
```

Comment:
```suggestion
    return NODAL_VAUX;
```

Reply:
I tried in this way before and it didn't work... I guess that it was creating a copy of the variable or something similar. This is why I retrieve it from the KratosComponent.s